### PR TITLE
Initial Mock class extraction

### PR DESCRIPTION
**Pull request by gildub**
The Mock class has been taken from fog-core lib code in order
to decouple the real code from the test code.

### CODE REVIEW

File: lib/fog/mock/errors.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+module Fog
+  module Errors
+    class MockNotImplemented < Fog::Errors::Error; end
+  end
+end
```
Comment (plribeiro3000) [inline lib/fog/mock/errors.rb:5]: Wouldn't the path of this file be `lib/fog/errors/mock_not_implemented` instead?

Comment (gildub) [reply lib/fog/mock/errors.rb:5]: @plribeiro3000, 

The file is about the module not each Error class.

Comment (plribeiro3000) [reply lib/fog/mock/errors.rb:5]: Still. There is only one constant being defined and the file name does not follow it.
It is a good practice to make the path follow the constant name to make it easier to find a class in the project.

Comment (gildub) [reply lib/fog/mock/errors.rb:5]: I understand the practice and I believe I'm respecting it. Here the path is `fog/mock/errors.rb` for a `Fog::Mock::Errors` module. This module contains only one class for now. Let me put it this way: we don't want to create a file for each Error class.

Comment (gildub) [reply lib/fog/mock/errors.rb:5]: This is irrelevant since this one is going back on the fog-core side.

**Comment by gildub** [comment]
@plribeiro3000, closing this one since it's not likely we're going to take the fog-mock gem approach no matter what we do with mock.

**Comment by plribeiro3000** [comment]
agreed.

**Pull request closed without merging**